modules/centroBidAdapter.js

```diff
@@ -72,23 +71,26 @@ var CentroAdapter = function CentroAdapter() {
    "value": 3.2,
    "adTag":''
    */
-  function _responseProcessing(resp, unit, placementCode) {
+  function _responseProcessing(resp, unit, requestedBid) {
     var bidObject;
     var bid = resp && resp.bid || resp;
 
-    if (bid && bid.adTag && bid.sectionID && bid.sectionID.toString() === unit.toString()) {
-      bidObject = bidfactory.createBid(1);
-      bidObject.cpm = bid.value;
-      bidObject.ad = bid.adTag;
-      bidObject.width = bid.width;
-      bidObject.height = bid.height;
+    if (bid && (bid.adTag || bid.statusMessage === 'No bid') && bid.sectionID && bid.sectionID.toString() === unit.toString()) {
+      if (bid.adTag) {
+        bidObject = bidfactory.createBid(1, requestedBid);
+        bidObject.cpm = bid.value;
+        bidObject.ad = bid.adTag;
+        bidObject.width = bid.width;
+        bidObject.height = bid.height;
+      } else {
+        bidObject = bidfactory.createBid(2, requestedBid);
+      }
     } else {
       // throw exception, or call utils.logError with resp.statusMessage
       utils.logError(LOG_ERROR_MESS.unitNum + unit + '. ' + (bid ? bid.statusMessage || LOG_ERROR_MESS.noAdTag : LOG_ERROR_MESS.noBid), bidderCode);
```

Comment:
It's ok. Main changes contains in this condition "(bid.adTag || bid.statusMessage === 'No bid')" logError must not called in case if bid.statusMessage === 'No bid' regardless of bid.adTag availability.